src/table/cells/__tests__/handle-cell-focus.spec.js

```diff
@@ -73,38 +73,112 @@ describe('handle-key-press', () => {
 
     beforeEach(() => {
       shouldRefocus = { current: false };
       hasSelections = false;
+      setfocusedCellCoord = sinon.spy();
     });
 
     it('should set tabindex on the first cell and not focus', () => {
-      handleCellFocus.handleResetFocus(focusedCellCoord, rootElement, shouldRefocus, hasSelections);
+      handleCellFocus.handleResetFocus({
+        focusedCellCoord,
+        rootElement,
+        shouldRefocus,
+        hasSelections,
+        setfocusedCellCoord,
+      });
       expect(cell.focus).to.not.have.been.called;
       expect(cell.blur).to.have.been.calledOnce;
       expect(cell.setAttribute).have.been.calledTwice;
+      expect(setfocusedCellCoord).to.have.been.calledOnceWith(focusedCellCoord);
     });
 
     it('should set tabindex on the first cell and focus when shouldRefocus is true', () => {
       shouldRefocus.current = true;
 
-      handleCellFocus.handleResetFocus(focusedCellCoord, rootElement, shouldRefocus, hasSelections);
+      handleCellFocus.handleResetFocus({
+        focusedCellCoord,
+        rootElement,
+        shouldRefocus,
+        hasSelections,
+        setfocusedCellCoord,
+      });
       expect(cell.focus).to.have.been.calledOnce;
       expect(cell.blur).to.have.been.calledOnce;
       expect(cell.setAttribute).have.been.calledTwice;
+      expect(setfocusedCellCoord).to.have.been.calledOnceWith(focusedCellCoord);
     });
 
     it('should set tabindex on the first cell in currently focused column when hasSelections is true', () => {
-      focusedCellCoord = { current: [0, 1] };
+      focusedCellCoord = [0, 1];
       rootElement = { getElementsByClassName: () => [{ getElementsByClassName: () => [cell, cell] }] };
       hasSelections = true;
 
-      handleCellFocus.handleResetFocus(focusedCellCoord, rootElement, shouldRefocus, hasSelections);
+      handleCellFocus.handleResetFocus({
+        focusedCellCoord,
+        rootElement,
+        shouldRefocus,
+        hasSelections,
+        setfocusedCellCoord,
+      });
+
       expect(cell.focus).to.not.have.been.called;
       expect(cell.blur).to.have.been.calledOnce;
       expect(cell.setAttribute).have.been.calledTwice;
-      expect(focusedCellCoord.current).to.eql([0, 1]);
+      expect(setfocusedCellCoord).to.have.been.calledOnceWith(focusedCellCoord);
+      expect(focusedCellCoord).to.eql([0, 1]);
+    });
+  });
+
+  describe('handleNavigateTop', () => {
+    let scrollTo;
+    let tableSection;
+    let rowHeight;
+
+    beforeEach(() => {
+      rowHeight = 100;
+      scrollTo = sinon.spy();
+      tableSection = { current: { scrollTo } };
+    });
+
+    it('should do any thing when ref is not setup yet', () => {
+      tableSection.current = {};
+
+      handleCellFocus.handleNavigateTop({ tableSection, focusedCellCoord, rootElement });
+      expect(scrollTo).to.not.have.been.called;
+    });
+
+    it('should scroll to top of the element when it dedetects cursor is behind `<TableHead />`', () => {
```

Comment:
detects

Reply:
✅ Fixed

Reply:
Oh actually, this one should say something like: `should scroll to the top when you reach the top two rows`

Reply:
✅ Fixed